test/core/test/expect.test.ts

```diff
@@ -40,3 +41,236 @@ describe('expect.soft', () => {
+    for (const obj of array)
+      yield obj
+  }
+
+  const customObject1 = createAnagramComparator('listen')
+  const customObject2 = createAnagramComparator('silent')
+
+  expect.addEqualityTesters([areObjectsEqual])
+
+  test('AnagramComparator objects are unique and not contained within arrays of AnagramComparator objects', () => {
+    expect(customObject1).not.toBe(customObject2)
+    expect([customObject1]).not.toContain(customObject2)
+  })
+
+  test('basic matchers pass different AnagramComparator objects', () => {
+    expect(customObject1).toEqual(customObject2)
+    expect([customObject1, customObject2]).toEqual([customObject2, customObject1])
+    expect(new Map([['key', customObject1]])).toEqual(new Map([['key', customObject2]]))
+    expect(new Set([customObject1])).toEqual(new Set([customObject2]))
+    expect(toIterator([customObject1, customObject2])).toEqual(
+      toIterator([customObject2, customObject1]),
+    )
+    expect([customObject1]).toContainEqual(customObject2)
+    expect({ a: customObject1 }).toHaveProperty('a', customObject2)
+    expect({ a: customObject2, b: undefined }).toStrictEqual({
+      a: customObject1,
+      b: undefined,
+    })
+    expect({ a: 1, b: { c: customObject1 } }).toMatchObject({
+      a: 1,
+      b: { c: customObject2 },
+    })
+  })
+
+  test('asymmetric matchers pass different AnagramComparator objects', () => {
+    expect([customObject1]).toEqual(expect.arrayContaining([customObject1]))
+    expect({ a: 1, b: { c: customObject1 } }).toEqual(
+      expect.objectContaining({ b: { c: customObject2 } }),
+    )
+  })
+
+  test('toBe recommends toStrictEqual even with different objects', () => {
+    expect(() => expect(customObject1).toBe(customObject2)).toThrow('toStrictEqual')
+  })
+
+  test('toBe recommends toEqual even with different AnagramComparator objects', () => {
+    expect(() => expect({ a: undefined, b: customObject1 }).toBe({ b: customObject2 })).toThrow(
+      'toEqual',
+    )
+  })
+
+  test('iterableEquality still properly detects cycles', () => {
+    const a = new Set()
+    a.add(customObject1)
+    a.add(a)
+
+    const b = new Set()
+    b.add(customObject2)
+    b.add(b)
+
+    expect(a).toEqual(b)
+  })
+})
+
+describe('recursive custom equality tester', () => {
+  let personId = 0
+
+  class Address {
+    public address: string
+
+    constructor(address: string) {
+      this.address = address
+    }
+  }
+  class Person {
+    public name: string
+    public address: Address
+    public personId: string
+
+    constructor(name: string, address: Address) {
+      this.name = name
+      this.address = address
+      this.personId = `${personId++}`
+    }
+  }
+
+  const arePersonsEqual = (a: unknown, b: unknown, customTesters: Array<Tester>) => {
+    const isAPerson = a instanceof Person
+    const isBPerson = b instanceof Person
+
+    if (isAPerson && isBPerson)
+      return a.name === b.name && equals(a.address, b.address, customTesters)
```

Comment:
Tests look good!

Just this one, I think using `this.equals` reflects API better than using `equals` export. Maybe you should rewrite it to `function arePersonsEqual(this: TesterContext, a: unknown, ...) { ... }`?